mission/functions/core/ui/taskroster/fn_tr_missions_fill.sqf

```diff
@@ -40,54 +34,142 @@ vn_mf_fnc_getTaskID =
-		//Don't add, if Mission is in a different Zone.
-		private _zoneName = _task_data#0;
-		
-		private _lnbindex = _ctrlSupportTasks lnbAddRow ["","","",(taskDescription _task)#1];
-		_ctrlSupportTasks lnbSetPicture					[ [_lnbindex, 1], getText(configFile >> "CfgTaskTypes" >> taskType _task >> "icon")];
-		
-		//store the taskname (if clicked on "Current Tasking" -> Select Index of mission)
-		_ctrlSupportTasks lnbSetData [[_lnbindex, 1], str(_task)];
-		
-		//get Task category
-		private _task_category = getText(missionConfigFile >> "gamemode" >> "tasks" >> _task_classname >> "taskcategory");
-		//Backup: Colors: VN_TR_MISS_PRIM - VN_TR_MISS_SECO - VN_TR_MISS_SUPP
-		private _typeData = switch(_task_category)do
-		{
-			case "PRI":	{[0, "\vn\ui_f_vietnam\ui\taskroster\img\icons\vn_icon_task_primary.paa"]};
-			case "SEC":	{[50, "\vn\ui_f_vietnam\ui\taskroster\img\icons\vn_icon_task_secondary.paa"]};
-			case "SUP":	{[100, "\vn\ui_f_vietnam\ui\taskroster\img\icons\vn_icon_task_support.paa"]};
-			default {[1, "Vn\ui_f_vietnam\ui\debrief\sticky.paa"]};
-		};
-		private _orderValue = (_typeData#0)+_forEachIndex;
-		_ctrlSupportTasks lnbSetValue					[ [_lnbindex, 0],	_orderValue];	//order value - used for sorting by Mission Type
-		_ctrlSupportTasks lnbSetPicture					[ [_lnbindex, 0],	(_typeData#1)];
-		_ctrlSupportTasks lnbSetPictureColor 			[ [_lnbindex, 0],	vn_mf_tr_zone_active_cur_color];
-		_ctrlSupportTasks lnbSetPictureColorSelected	[ [_lnbindex, 0],	vn_mf_tr_zone_active_cur_color];
-		//Set the data for missions_fill, to be loaded in the mission-Infosheet
-		vn_tr_taskList pushback [_orderValue, _task_category, _task_classname, _task];
-	};
-}forEach simpleTasks player;
+	private _t = [_x, _forEachIndex] call _fnc_process_task;
+	vn_tr_taskList pushBack (_t get "tasklistClientVarData");
+
+	// new row of data
+	// **MUST** set at least one value in here otherwise it breaks the UI
+	// and refuses to show the right hand side page.
+	private _row = _ctrl lnbAddRow ["", "", "", "", _t get "taskDescription"];
+
+	// set the task data. this is grabbed by `vn_mf_fnc_tr_missions_show` when
+	// activating the currently selected task on the right hand side page.
+	_ctrl lnbSetData [[_row, 1], _t get "taskData"];
+
+	// column 1 -- icon for task type
+	_ctrl lnbSetValue [[_row, 0], _t get "taskSortOrderRank"];
+	_ctrl lnbSetPicture [[_row, 0], _t get "taskTypeIcon"];
+	_ctrl lnbSetPictureColor [[_row, 0], _t get "taskTypeIconColor"];
+	_ctrl lnbSetPictureColorSelected [[_row, 0], _t get "taskTypeIconColor"];
+
+	// column 2 -- text for task type
+	_ctrl lnbSetText [[_row, 1], _t get "taskTypeText"];
+
+	// column 3 -- icon for task subtype
+	_ctrl lnbSetPicture [[_row, 2], _t get "taskSubtypeIcon"];
+	_ctrl lnbSetPictureColor [[_row, 2], _t get "taskTypeIconColor"];
+	_ctrl lnbSetPictureColorSelected [[_row, 2], _t get "taskTypeIconColor"];
+
+	// column 4 -- text for task subtype
+	// TODO: Not showing for some reason.
```

Comment:
!todo -- pretty sure this is to do with the percentages that need editing in configs